compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -141,14 +141,18 @@ impl<'a> ModCollector<'a> {
     fn collect_impls(&mut self, context: &mut Context, impls: Vec<TypeImpl>, krate: CrateId) {
         let module_id = ModuleId { krate, local_id: self.module_id };
 
-        for r#impl in impls {
+        for mut r#impl in impls {
```

Comment:
```suggestion
        for r#impl in impls {
```
Don't need this any more

Reply:
I had to move this mut to the other loop:
```rust
for (mut method, _) in r#impl.methods {
```